File: py-rsmdu/examples/iris_from_ign.py
"""
Example: Download IRIS (statistical units) data from IGN API using rsmdu

This example demonstrates how to:
1. Create an Iris instance
2. Set a bounding box
3. Download IRIS from IGN API
4. Get GeoJSON data
5. Save to GPKG file
"""

import json

import rsmdu


def main():
    print("📊 Loading IRIS from IGN API...")

    # Create Iris instance
    iris = rsmdu.geometric.Iris(output_path="./output")

    # Set bounding box (La Rochelle area, France)
    # Format: min_x, min_y, max_x, max_y (WGS84, EPSG:4326)
    iris.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699)

    # Set CRS (optional, defaults to EPSG:2154)
    iris.set_crs(2154)

    geo = iris.geo_core
    print(f"📦 Bounding box set")
    print(f"🗺️  CRS: {geo.epsg}")
    print(f"📁 Output path: {geo.output_path}")

    # Run iris processing: downloads from IGN API and parses GeoJSON
    print("⏳ Downloading IRIS from IGN API...")
    iris = iris.run()

    # Get GeoJSON (equivalent to to_gdf() in Python)
    print("📊 Getting GeoJSON data...")
    geojson = iris.get_geojson()

    if geojson and "features" in geojson:
        num_features = len(geojson["features"])
        print(f"✅ Loaded {num_features} IRIS units")
    else:
        print("✅ IRIS data loaded")

    # Save to GPKG
    print("💾 Saving to GPKG...")
    iris.to_gpkg(name="iris")

    print(f"✅ IRIS processing complete!")
    print(f"📁 Output path: {iris.get_output_path()}")

    return iris


if __name__ == "__main__":
    iris = main()

File: rsmdu/examples/iris_from_ign.rs
use anyhow::Result;
use rsmdu::geometric::iris::Iris;

/// Example: Loading IRIS (statistical units) data from IGN API
/// Following Python example from pymdu.geometric.Iris
fn main() -> Result<()> {
    println!("=== Example: Loading IRIS from IGN API ===\n");

    // Create Iris instance
    // Python: iris = Iris(output_path='./')
    let mut iris = Iris::new(Some("./output".to_string()))?;

    // Set bounding box (La Rochelle, France)
    // Python: iris.bbox = [-1.152704, 46.181627, -1.139893, 46.18699]
    iris.set_bbox(-1.152704, 46.181627, -1.139893, 46.18699);

    println!("Bounding box set:");
    println!("  - Longitude: -1.152704 to -1.139893");
    println!("  - Latitude: 46.181627 to 46.18699");
    println!("  - Area: La Rochelle, France");
    println!("  - Format: WGS84 (EPSG:4326)\n");

    // Run iris processing
    // Python: iris = iris.run()
    println!("Downloading and processing IRIS from IGN API...");
    let iris_result = iris.run()?;

    println!("\nIRIS processed successfully!");

    // Get GeoJSON (equivalent to to_gdf() in Python)
    if let Some(geojson) = iris_result.get_geojson() {
        match geojson {
            geojson::GeoJson::FeatureCollection(fc) => {
                println!("  - Number of IRIS units: {}", fc.features.len());
            }
            _ => {
                println!("  - GeoJSON loaded (non-FeatureCollection format)");
            }
        }
    }

    // Save to GeoJSON
    // Python: iris.to_geojson(name="iris")
    println!("\nSaving to GeoJSON...");
    iris_result.to_geojson(None)?;

    println!("\n✅ Processing complete!");
    println!("  - Output file: {:?}", iris_result.get_output_path());

    Ok(())
}
